README.md

```diff
@@ -87,3 +87,129 @@ An ESX host can be attached and managed indirectly via vSphere API:
     }
 
 See tests folder for additional examples.
+
+## ESXi resource types
+### esx_advanced_options
+#### Parameters
+* `transport`: A resource reference to a transport type declared elsewhere. Eg: `Transport['vcenter']`
+* `options`: A hash containing a list of options:
+```
+options => {
+"Vpx.Vpxa.config.log.level"  => "verbose",   # ChoiceOption  default "verbose"
+"Config.HostAgent.log.level" => "verbose",   # ChoiceOption  default "verbose"
+"Annotations.WelcomeMessage" => "",          # StringOption  default ""
+"BufferCache.SoftMaxDirty"   => 15,          # LongOption    default 15
+"CBRC.Enable"                => false,       # BoolOption    default false
+"Config.GlobalSettings.guest.commands.sharedPolicyRefCount" => 0   # IntOption     default 0
+```
+
+### esx_datastore
+#### Parameters
+* `ensure`: present
+* `type`: 'NFS,CIFS,VMFS'
+* `remote_host`: IP or DNS name of remote host.
+* `remote_path`: Path to directory/folder or remote host.
+* `transport`: A resource reference to a transport type declared elsewhere. Eg: `Transport['vcenter']`
```

Comment:
Another one where params are set based off resource name. host, datastore and local_path all set using the format "host:datastore_name"

https://github.com/vmware/vmware-vcenter/blob/master/lib/puppet/type/esx_datastore.rb#L5-L14